### DISCUSSION TITLE
Dual task target

### DISCUSSION THREAD

**Discussion by fsikora** [ds4dm/ml4co-competition#76] [Q&A]
Hi all,
We're wondering how the dual task should be understood regarding to the solution value. 
I've read here that you didn't provide optimal value for the training instances. 
Would it be the case for the testing part ? 
If not, how the dual task is supposed to stop then?

Thanks,
Florian

**Comment by gasse** [comment]
Hi @fsikora ,

The dual bound is basically the lowest LP solution value, among all leaf nodes in the B&B tree. We do not provide the optimal solution value to SCIP indeed, simply because for some instances it was unrealistic to compute. Instead we impose a time limit to process each instance during the evaluation, which is 15 minutes for the dual task. So either SCIP can solve the instance within the time limit and stops early (which can happen in the dual task, although it is unlikely since we deactivate all primal heuristics), or SCIP reaches the time limit and stops. In both cases, our computation of the reward functions (integral over the whole time window) is supposed to handle those case properly.

Let us know if you need more clarifications.

Best,
Maxime
